packages/checkout-storefront/src/sections/PaymentSection/AdyenDropIn.tsx

```diff
@@ -0,0 +1,193 @@
+        case AdyenPaymentResponse.ResultCodeEnum.IdentifyShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Pending:
+        case AdyenPaymentResponse.ResultCodeEnum.PresentToShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Received:
+        case AdyenPaymentResponse.ResultCodeEnum.RedirectShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Refused: {
+          console.error(result);
+          component.setStatus("error", {
+            message: `${result.payment.resultCode}: ${result.payment.refusalReason}`,
+          });
+          return;
+        }
+
+        case AdyenPaymentResponse.ResultCodeEnum.Authorised:
+        case AdyenPaymentResponse.ResultCodeEnum.Success: {
+          const newUrl = `?order=${result.orderId}`;
+          window.location.href = newUrl;
+          return;
+        }
+      }
+    }
+  });
+
+  const onAdditionalDetails: AdyenCheckoutInstanceOnAdditionalDetails = useEvent(
+    (_state, _component) => {
+      // @todo Your function calling your server to make a `/payments/details` request
+    }
+  );
+
+  useEffect(() => {
+    if (!dropinContainerElRef.current || !adyenSessionResponse.data) {
+      return;
+    }
+
+    createAdyenCheckoutInstance(adyenSessionResponse.data, { onSubmit, onAdditionalDetails })
+      .then((adyenCheckout) => {
+        setAdyenCheckoutInstance(adyenCheckout);
+        dropinComponentRef.current = adyenCheckout
+          .create("dropin")
+          .mount(dropinContainerElRef.current!);
+      })
+      .catch(console.error);
+
+    return () => {
+      dropinComponentRef.current?.unmount();
+    };
+  }, [adyenSessionResponse.data]);
+
+  return <div ref={dropinContainerElRef} />;
+});
+AdyenDropIn.displayName = "AdyenDropIn";
+
+function createAdyenCheckoutInstance(
+  adyenSessionResponse: AdyenDropInCreateSessionResponse,
+  {
+    onSubmit,
+    onAdditionalDetails,
+  }: {
+    onSubmit: AdyenCheckoutInstanceOnSubmit;
+    onAdditionalDetails: AdyenCheckoutInstanceOnAdditionalDetails;
+  }
+) {
+  return AdyenCheckout({
+    environment: "test",
+    clientKey: adyenSessionResponse.clientKey,
+    session: {
+      id: adyenSessionResponse.session.id,
+      sessionData: adyenSessionResponse.session.sessionData,
+    },
+    onPaymentCompleted: (result: any, component: any) => {
+      console.info(result, component);
+    },
+    onError: (error: any, component: any) => {
+      console.error(error.name, error.message, error.stack, component);
+    },
+    onSubmit,
+    onAdditionalDetails,
+    // Any payment method specific configuration. Find the configuration specific to each payment method: https://docs.adyen.com/payment-methods
+    // For example, this is 3D Secure configuration for cards:
+    paymentMethodsConfiguration: {
+      card: {
+        hasHolderName: true,
+        holderNameRequired: true,
+        billingAddressRequired: false,
+      },
+    },
+    analytics: {
+      enabled: false,
+    },
+  });
+}
```

Comment:
I think we should extract this function into a separate file - this file is already pretty... intense :P